apps/astarte_housekeeping_api/lib/astarte_housekeeping_api/config.ex

```diff
@@ -17,28 +17,60 @@
 #
 
 defmodule Astarte.Housekeeping.API.Config do
-  @moduledoc """
-  This module contains functions to access the configuration
-  """
+  use Skogsra
 
-  @doc """
-  Returns true if the authentication is disabled
-  """
-  def jwt_public_key_pem do
-    Application.get_env(:astarte_housekeeping_api, :jwt_public_key_pem)
-  end
+  @envdoc "The path to the public key used to verify the JWT auth."
+  app_env :jwt_public_key_path, :astarte_housekeeping_api, :jwt_public_key_path,
+    os_env: "HOUSEKEEPING_API_JWT_PUBLIC_KEY_PATH",
+    type: :binary,
+    default: ""
 
-  @doc """
-  Returns true if the authentication is disabled
+  @envdoc "The port used from the Phoenix server."
+  app_env :port, :astarte_housekeeping_api, :port,
+    os_env: "PORT",
+    type: :integer,
+    default: 4001
+
+  @envdoc "The bind address for the Phoenix server."
+  app_env :bind_address, :astarte_housekeeping_api, :bind_address,
+    os_env: "HOUSEKEEPING_API_BIND_ADDRESS",
+    type: :binary,
+    default: "0.0.0.0"
+
+  @envdoc """
+  Disables the authentication. CHANGING IT TO TRUE IS GENERALLY A REALLY BAD IDEA IN A PRODUCTION ENVIRONMENT, IF YOU DON'T KNOW WHAT YOU ARE DOING.
   """
-  def authentication_disabled? do
-    Application.get_env(:astarte_housekeeping_api, :disable_authentication, false)
+  app_env :disable_authentication, :astarte_housekeeping_api, :disable_authentication,
+    os_env: "HOUSEKEEPING_API_DISABLE_AUTHENTICATION",
+    type: :atom,
+    default: false
+
+  @doc "The RPC client module."
+  app_env :rpc_client, :astarte_housekeeping_api, :rpc_client,
+    os_env: "HOUSEKEEPING_API_RPC_CLIENT",
+    type: :atom,
+    default: Astarte.RPC.AMQP.Client
+
+  @doc "Returns the JWT public key."
+  def jwt_public_key_pem! do
+    public_key_path = jwt_public_key_path!()
+
+    cond do
+      authentication_disabled? ->
+        ""
+
+      public_key_path == "" ->
+        raise "No JWT public key path configured"
+
+      true ->
+        File.read(public_key_path)
+    end
   end
 
-  @doc """
-  Returns the RPC client module
+  @envdoc """
+  Returns true if the authentication is disabled.
   """
-  def rpc_client do
-    Application.get_env(:astarte_housekeeping_api, :rpc_client, Astarte.RPC.AMQP.Client)
+  def authentication_disabled? do
+    disable_authentication!()
```

Comment:
You're missing a `not` here

Reply:
No, `disable_authentication!()` simply returns the value of the env var.